File: src/main/java/control/BinaryChoiceAlertController.java
package control;

import app.AppSettings;
import javafx.scene.control.Button;
import javafx.stage.Stage;

import java.net.URL;
import java.util.ResourceBundle;

public class BinaryChoiceAlertController extends BasicAlertController {

    public Button rejectButton;

    public BinaryChoiceAlertController(String text){
        super(text);
    }

    public BinaryChoiceAlertController(String text, boolean defaultState){
        super(text, defaultState);
    }

    @Override
    public void initialize(URL url, ResourceBundle resourceBundle) {
        super.initialize(url, resourceBundle);

        rejectButton.setText(AppSettings.language().getItem("denyButton"));
        //acceptButton.setText(AppSettings.language().getItem("acceptButton"));

        rejectButton.setOnAction(event -> {
            accepted = false;
            event.consume();
            ((Stage) rejectButton.getScene().getWindow()).close();
        });
    }
}

File: src/main/scala/control/BinaryChoiceAlertController.scala
package control

import java.net.URL
import java.util.ResourceBundle

import app.AppSettings
import javafx.fxml.FXML
import javafx.scene.control.Button
import javafx.stage.Stage

class BinaryChoiceAlertController(text: String, defaultState: Boolean = false)
    extends BasicAlertController(text, defaultState) {

    @FXML var rejectButton: Button = _

    //For Java interoperability transparency
    def this(text: String) = this(text, false)

    override def initialize(url: URL, resourceBundle: ResourceBundle): Unit = {
        super.initialize(url, resourceBundle)

        rejectButton.setText(AppSettings.language.getItemOrElse("denyButton", "Reject"))
        //acceptButton.setText(AppSettings.language().getItemOrElse("acceptButton", "Accept"));

        rejectButton.setOnAction(event => {
            _accepted = false
            event.consume()
            rejectButton.getScene.getWindow.asInstanceOf[Stage].close()
        })
    }
}